frontend/src/Features/Search/Search.tsx

```diff
@@ -1,93 +1,148 @@
+  {id: '5', name: 'Mason', surname: 'Taylor', profilePicture: 'https://via.placeholder.com/40', login: 'mason'},
+  {id: '6', name: "Jan", surname: 'Kowalski', profilePicture: 'https://via.placeholder.com/40', login: 'johndoe'},
 ];
 
 const usersData = users.reduce((acc, user) => {
-  acc[user.name] = {avatar: user.avatar, tag: user.tag};
+  acc[`${user.name} ${user.surname}`] = {profilePicture: user.profilePicture, tag: `@${user.login}`};
   return acc;
-}, {} as Record<string, { avatar: string; tag: string }>);
-
-const renderAutocompleteOption: AutocompleteProps['renderOption'] = ({option}) => (
-  <Group gap="sm">
-    <Avatar src={usersData[option.value].avatar} size={36} radius="xl"/>
-    <div>
-      <Text size="sm">{option.value}</Text>
-      <Text size="xs" opacity={0.5}>
-        {usersData[option.value].tag}
+}, {} as Record<string, { profilePicture: string | null; tag: string }>);
+
+const renderAutocompleteOption: AutocompleteProps['renderOption'] = ({option}) => {
+  if (option.value.startsWith("search:")) {
+    const searchText = option.value.replace("search:", "");
+    return (
+      <Text size="sm" opacity={0.7}>
+        Wyszukaj "{searchText}"
       </Text>
-    </div>
-  </Group>
-);
+    );
+  }
+
+  return (
+    <Group gap="sm">
+      <Avatar src={usersData[option.value]?.profilePicture} size={36} radius="xl"/>
+      <div>
+        <Text size="sm">{option.value}</Text>
+        <Text size="xs" opacity={0.5}>
+          {usersData[option.value]?.tag}
+        </Text>
+      </div>
+    </Group>
+  );
+};
 
 export const Search = () => {
   const [query, setQuery] = useState('');
+  const [dropdownOpened, setDropdownOpened] = useState(false);
   const navigate = useNavigate();
+  const itemsNumber = 5;
 
   const convertedData = users
     .map((user) => ({
-      value: user.name,
-      label: user.name,
-      tag: user.tag,
-      avatar: user.avatar,
+      value: `${user.name} ${user.surname}`,
+      label: `${user.name} ${user.surname}`,
+      tag: `@${user.login}`,
+      profilePicture: user.profilePicture,
     }));
 
-  // Dane do Autocomplete (nazwy użytkowników i tagi użytkowników)
+  // Data for autocomplete options
   const optionsFilter: OptionsFilter = ({options, search}) => {
-
     if (!search) {
       return options;
     }
 
-    return (options as ComboboxItem[]).filter((option) => {
+    const filteredOptions = (options as ComboboxItem[]).filter((option) => {
       const searchLower = search.toLowerCase();
-      return option.value.toLowerCase().includes(searchLower) || usersData[option.value].tag.toLowerCase().includes(searchLower);
+      return option.value.toLowerCase().includes(searchLower) || usersData[option.value]?.tag.toLowerCase().includes(searchLower);
     });
+
+    // Add dynamic option for searching users
+    return [
+      {
+        value: `search:${search}`,
```

Comment:
One starts with capital letter, other with lowercase. Is it correct?

Reply:
Yup it is. search: is used for query operations and this "Wyszukaj" is just a label which is visible during searching for users.

Reply:
okie, no more questions